public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/PyroscopeAppIntegration.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useEffect, useMemo, useState} from 'react';
+
+import { QueryEditorProps, TimeRange, DateTime } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import {LinkButton} from '@grafana/ui'
+
+import { PyroscopeDataSource } from '../datasource';
+import { PyroscopeDataSourceOptions, Query } from '../types';
+
+const PYROSCOPE_APP_ID = 'grafana-pyroscope-app'
+
+/* Global promise to fetch the pyroscope app settings */
+let pyroscopeAppSettings: Promise<PyroscopeAppSettings> | null= null;
```

Comment:
Why do we have global promise ref here? I don't think multiple instances of the editor should share this promise.

Reply:
I only want to fetch the plugin settings exactly once per browser refresh.
Same with the settings for each datasource.

If there was a change to any of these during the user's active browser state, they will have needed to refresh anyway to take advantage of them.

Reply:
Ah sorry ok it's the App settings and there should be only one instance for this. Yeah in that case this makes sense.